planner/core/planbuilder.go

```diff
@@ -1218,13 +1277,34 @@ func (b *PlanBuilder) buildPhysicalIndexLookUpReaders(ctx context.Context, dbNam
 	// get index information
 	indexInfos := make([]*model.IndexInfo, 0, len(tblInfo.Indices))
 	indexLookUpReaders := make([]Plan, 0, len(tblInfo.Indices))
+
+	check := b.isForUpdateRead && b.ctx.GetSessionVars().ConnectionID > 0
+	var latestIndexes map[int64]*model.IndexInfo
+	var err error
+	if b.isForUpdateRead && b.ctx.GetSessionVars().ConnectionID > 0 {
+		latestIndexes, err = getLatestIndexInfo(b.ctx, tblInfo.ID, b.is.SchemaMetaVersion())
+		if err != nil {
+			return nil, nil, err
+		}
+		if latestIndexes == nil {
+			check = false
+		}
+	}
+
 	for _, idx := range indices {
 		idxInfo := idx.Meta()
 		if idxInfo.State != model.StatePublic {
 			logutil.Logger(context.Background()).Info("build physical index lookup reader, the index isn't public",
 				zap.String("index", idxInfo.Name.O), zap.Stringer("state", idxInfo.State), zap.String("table", tblInfo.Name.O))
 			continue
 		}
+		if check {
+			if latestIndex, ok := latestIndexes[idxInfo.ID]; !ok || latestIndex.State != model.StatePublic {
+				logutil.BgLogger().Info("build physical index lookup reader, the index isn't public in forUpdateRead",
```

Comment:
Use `logutil.Logger(ctx)` instead or if this is a debug log we could remove it.

Reply:
There are similar log in this function, I think they can both be removed.